### ISSUE TITLE
dgl.random.seed initialize CUDA context unexpectedly

### ISSUE THREAD

**Issue by skepsun** [dmlc/dgl#3054]
## 🐛 Bug

I have a machine with multiple GPUs, an extra memory cost (around 309Mb) appears in GPU 0 when running scripts specifying cuda device to GPU n (n>0).

## To Reproduce
Debug the following codes while observing GPU usage using nvidia-smi
```
import dgl
import torch
torch.cuda.set_device("cuda:1")
dgl.random.seed(0)
# set breakpoint here
print()
```

## Expected behavior

Both GPU 0 and GPU 1 will cost extra 309Mb memory.

## Environment

 - DGL Version (e.g., 1.0): 0.6.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): 1.8.1+cu101
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source): 
 - Python version: 3.8.8
 - CUDA/cuDNN version (if applicable): 10.1
 - GPU models and configuration (e.g. V100): V100 * 8
 - Any other relevant information:

**Comment by VoVAllen** [comment]
This is expected. If you try 
```python
import torch
a = th.randn((1,1), device="cuda:1")
```
Output
```
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA Tesla T4     On   | 00000000:00:1C.0 Off |                    0 |
| N/A   32C    P0    25W /  70W |   1158MiB / 15109MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

You'll see torch occupied more than 1000MB in GPU memory.

**Comment by skepsun** [comment]
> This is expected. If you try
> 
> ```python
> import torch
> a = th.randn((1,1), device="cuda:1")
> ```
> 
> Output
> 
> ```
> +-------------------------------+----------------------+----------------------+
> |   1  NVIDIA Tesla T4     On   | 00000000:00:1C.0 Off |                    0 |
> | N/A   32C    P0    25W /  70W |   1158MiB / 15109MiB |      0%      Default |
> |                               |                      |                  N/A |
> +-------------------------------+----------------------+----------------------+
> ```
> 
> You'll see torch occupied more than 1000MB in GPU memory.

But I do not want to occupy extra memory in GPU: 0 (the first available device) if I have set cuda device to other GPU cards (GPU: 1 or 2 ... or 7).

**Comment by strawhatboy** [comment]
seems that I encountered this issue too when cuda:0's memory is almost full, currently using:
* python=3.8
* dgl-cuda10.1=0.7.1

tried to revert dgl version to 0.6.1 but still the same issue
by reverting to 0.6.0, dgl.seed works fine on the correct GPU device.
